desktop/context.go

```diff
@@ -0,0 +1,191 @@
+	// ModelRunnerEngineKindMobyManual represents a non-Desktop/Cloud engine
+	// that's explicitly targeted by a DMR_HOST environment variable on which
+	// the user is responsible for managing a Model Runner.
+	ModelRunnerEngineKindMobyManual
+	// ModelRunnerEngineKindDesktop represents a Docker Desktop engine. It only
+	// refers to a Docker Desktop Linux engine, i.e. not a Windows container
+	// engine in the case of Docker Desktop for Windows.
+	ModelRunnerEngineKindDesktop
+	// ModelRunnerEngineKindCloud represents a Docker Cloud engine.
+	ModelRunnerEngineKindCloud
+)
+
+// String returns a human-readable engine kind description.
+func (k ModelRunnerEngineKind) String() string {
+	switch k {
+	case ModelRunnerEngineKindMoby:
+		return "Moby"
+	case ModelRunnerEngineKindMobyManual:
+		return "Moby (Manual Install)"
+	case ModelRunnerEngineKindDesktop:
+		return "Docker Desktop"
+	case ModelRunnerEngineKindCloud:
+		return "Docker Cloud"
+	default:
+		return "Unknown"
+	}
+}
+
+// ModelRunnerContext encodes the operational context of a Model CLI command and
+// provides facilities for inspecting and interacting with the Model Runner.
+type ModelRunnerContext struct {
+	// kind stores the associated engine kind.
+	kind ModelRunnerEngineKind
+	// urlPrefix is the prefix URL for all requests.
+	urlPrefix *url.URL
+	// client is the model runner client.
+	client DockerHttpClient
+}
+
+// DetectContext determines the current Docker Model Runner context.
+func DetectContext(cli *command.DockerCli) (*ModelRunnerContext, error) {
+	// Check for an explicit endpoint setting.
+	dmrHost := os.Getenv("DMR_HOST")
+
+	// Detect the associated engine type.
+	kind := ModelRunnerEngineKindMoby
+	if dmrHost != "" {
+		kind = ModelRunnerEngineKindMobyManual
+	} else if isDesktopContext(cli) {
+		kind = ModelRunnerEngineKindDesktop
+	} else if isCloudContext(cli) {
+		kind = ModelRunnerEngineKindCloud
+	}
+
+	// Compute the URL prefix based on the associated engine kind.
+	var rawURLPrefix string
+	if kind == ModelRunnerEngineKindMoby {
+		rawURLPrefix = "http://localhost:12434"
+	} else if kind == ModelRunnerEngineKindMobyManual {
+		rawURLPrefix = dmrHost
+	} else if kind == ModelRunnerEngineKindDesktop {
+		rawURLPrefix = "http://localhost" + inference.ExperimentalEndpointsPrefix
+	} else {
+		rawURLPrefix = "http://localhost/"
+	}
+	urlPrefix, err := url.Parse(rawURLPrefix)
+	if err != nil {
+		return nil, fmt.Errorf("invalid model runner URL (%s): %w", rawURLPrefix, err)
+	}
+
+	// Construct the HTTP client.
+	var client DockerHttpClient
+	if kind == ModelRunnerEngineKindMoby || kind == ModelRunnerEngineKindMobyManual {
+		client = http.DefaultClient
+	} else if kind == ModelRunnerEngineKindDesktop {
+		dockerClient, err := dockerClientForContext(cli, "desktop-linux")
+		if err != nil {
+			return nil, fmt.Errorf("unable to create model runner client: %w", err)
+		}
+		client = dockerClient.HTTPClient()
+	} else if kind == ModelRunnerEngineKindCloud {
```

Comment:
```suggestion
	} else {
	    // ModelRunnerEngineKindCloud
```
to get rid of the linter `Condition is always true` warning?